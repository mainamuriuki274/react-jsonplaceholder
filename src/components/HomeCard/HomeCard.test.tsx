import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { HomeCard } from './index';

describe('Home Card Component', () => {
  it('it renders', () => {
    render(
      <MemoryRouter>
        <HomeCard link="/" title="Test Title" description="Test description" />
      </MemoryRouter>,
    );
    expect(screen.getByText('Test Title')).toBeInTheDocument();
    expect(screen.getByText('Test description')).toBeInTheDocument();
  });
  it('it matches snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <HomeCard
            link="/"
            title="Test Title"
            description="Test description"
          />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
