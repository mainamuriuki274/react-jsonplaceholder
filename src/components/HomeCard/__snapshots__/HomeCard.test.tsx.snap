// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Card Component it matches snapshot 1`] = `
<a
  className="home_card_container"
  href="/"
  onClick={[Function]}
>
  <div
    className="home_card card"
  >
    <div
      className="card-body"
    >
      <div
        className="home_card_title card-title h5"
      >
        Test Title
      </div>
      <p
        className="card-text"
      >
        Test description
      </p>
    </div>
    <div
      className="home_card_footer card-footer"
    >
      View Â 
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </div>
</a>
`;
